app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java

```diff
@@ -0,0 +1,576 @@
+
+    public Mono<ApplicationJson> exportApplicationById(String applicationId) {
+        ApplicationJson applicationJson = new ApplicationJson();
+        Map<String, String> pluginMap = new HashMap<>();
+        Map<String, String> datasourceMap = new HashMap<>();
+        Map<String, String> newPageIdMap = new HashMap<>();
+    
+        if (applicationId == null || applicationId.isEmpty()) {
+            return Mono.error(new AppsmithException(AppsmithError.INVALID_PARAMETER, FieldName.APPLICATION_ID));
+        }
+
+        return pluginRepository
+            .findAll()
+            .collectList()
+            .map(pluginList -> {
+                pluginList.forEach(plugin -> pluginMap.put(plugin.getId(), plugin.getPackageName()));
+                return pluginList;
+            })
+            .flatMap(ignore -> applicationService.findById(applicationId, AclPermission.READ_APPLICATIONS))
+            .switchIfEmpty(Mono.error(new AppsmithException(AppsmithError.NO_RESOURCE_FOUND, FieldName.APPLICATION, applicationId)))
+            .flatMap(application -> {
+                ApplicationPage defaultPage = application.getPages()
+                    .stream()
+                    .filter(ApplicationPage::getIsDefault)
+                    .findFirst()
+                    .orElse(null);
+
+                if (defaultPage == null) {
+                    return Mono.error(new AppsmithException(AppsmithError.NO_RESOURCE_FOUND, FieldName.DEFAULT_PAGE_NAME));
+                }
+                return Mono.zip(
+                    newPageRepository.findById(defaultPage.getId(), AclPermission.READ_PAGES),
+                    Mono.just(defaultPage),
+                    Mono.just(application)
+                );
+            })
+            .flatMap(tuple -> {
+                NewPage defaultPage = tuple.getT1();
+                ApplicationPage defaultPageRef = tuple.getT2();
+                Application application = tuple.getT3();
+
+                defaultPageRef.setId(defaultPage.getUnpublishedPage().getName());
+                final String organizationId = application.getOrganizationId();
+                application.setOrganizationId(null);
+                examplesOrganizationCloner.makePristine(application);
+                String applicationName = application.getName();
+                applicationJson.setExportedApplication(application);
+                return newPageRepository.findByApplicationId(applicationId, AclPermission.READ_PAGES)
+                    .collectList()
+                    .flatMap(newPageList -> {
+                        Map<String, Set<String>> mongoEscapedWidgetsNames = new HashMap<>();
+                        newPageList.forEach(newPage -> {
+                            newPageIdMap.put(newPage.getId(), newPage.getUnpublishedPage().getName());
+                            newPage.setApplicationId(applicationName);
```

Comment:
Are we using this `applicationName` anywhere else? Can we set the `applicationId` to null instead? When importing, the new application ID would be set inside the pages anyway right?

```suggestion
                            newPage.setApplicationId(null);
```

Reply:
Updated code pushed!